### PR TITLE
Use ListItem component

### PR DESCRIPTION
**Pull request by marcoambrosini**
This removes `AppContentListItem` and adds the upstreamed `ListItem` component.
requres https://github.com/nextcloud/nextcloud-vue/pull/616

Signed-off-by: Marco Ambrosini <marcoambrosini@pm.me>

### CODE REVIEW

**Comment by nickvergessen** [comment]
Jest / node12.x (pull_request) Failing after 1m — node12.x

File: src/components/LeftSidebar/ConversationsList/Conversation.vue [hunk 1]
```diff
@@ -40,12 +42,6 @@
 				{{ conversationInformation }}
 			</template>
 		</template>
-		<AppNavigationCounter v-if="item.unreadMessages"
-			slot="counter"
-			class="counter"
-			:highlighted="counterShouldBePrimary">
```
Comment (nickvergessen) [inline src/components/LeftSidebar/ConversationsList/Conversation.vue:46]: quite a big feature regression

Comment (marcoambrosini) [reply src/components/LeftSidebar/ConversationsList/Conversation.vue:46]: solved

File: src/components/LeftSidebar/ConversationsList/Conversation.vue [hunk 2]
```diff
@@ -20,11 +20,14 @@
...
+	<ListItem
 		:title="item.displayName"
 		:anchor-id="`conversation_${item.token}`"
-		:to="item.token ? { name: 'conversation', params: { token: item.token }} : ''"
-		:class="{ 'has-unread-messages': item.unreadMessages }"
+		:active="isActive"
+		:to="to"
+		:bold="!!item.unreadMessages"
+		:counter-number="item.unreadMessages"
+		:counter-highlighted="counterShouldBePrimary"
```
Comment (nickvergessen) [inline src/components/LeftSidebar/ConversationsList/Conversation.vue:30]: Invalid prop: type check failed for prop "counterHighlighted". Expected Boolean, got Number with value 0.

**Review by nickvergessen — COMMENTED** [review]
Looks good otherwise

File: src/components/LeftSidebar/ConversationsList/Conversation.vue (hunk 2 above)
Comment (marcoambrosini) [reply src/components/LeftSidebar/ConversationsList/Conversation.vue:30]: fixed

**Comment by nickvergessen** [comment]
> await listedEls.at(3).find('a').trigger('click')

Jest fails

**Comment by nickvergessen** [comment]
- [ ] See https://github.com/nextcloud/spreed/pull/5855 for a follow up patch that needs inclusion

**Review by PVince81 — APPROVED** [review]
Tested, works :+1:

**Pull request merged**